test/registryauth_test.go

```diff
@@ -47,7 +54,7 @@ func testValidEncodedAuthConfig(t *testing.T, user, password, imageName, expecte
 	}
 
 	if user != authConfig.Username {
-		t.Log("Auth config username mismatch", user, authConfig.Username)
+		t.Logf("Auth config username mismatch want %s, got: %s", user, authConfig.Username)
```

Comment:
Great way to improve the readability of this test. Do tests fail if you keep the test code but revert the code fix?

Reply:
The tests were passing with this line in it's original state, but I was having trouble knowing what the problem was, so I fixed it :smile:

Reply:
I just re-read your comment. I misunderstood you the first time around...
Yes, by only adding in the additional test cases, the tests would fail. In the original state of the code, the repository _must_ be included, and no tag could be used. This is due to the assumptions that the `WithName` function parsed the input (from the docker library)

Reply:
I was referring to the Logs usage / want/got values are really important in a test failure 👍